specifyweb/businessrules/tests/cog_type.py

```diff
@@ -0,0 +1,16 @@
+
+from specifyweb.businessrules.exceptions import BusinessRuleException
+from specifyweb.specify.models import CollectionObjectGroupType
+from specifyweb.specify.tests.test_api import ApiTests
+
+# NOTE: Edit this test when a new COG type rule is decided upon
```

Comment:
If you want to stand out, figure out why this test passed before (but I understand time constraints, so will not hold it against you if you don't)

Reply:
The tests were not passing because they were never running. 

The tests would have needed to have been added to https://github.com/specify/specify7/blob/production/specifyweb/businessrules/tests/__init__.py

This can can be verified by executing the following command:  
```bash
docker exec -it specify7-specify7-1 bash -c "ve/bin/python manage.py test specifyweb.businessrules.tests"
```

The `cog_type` tests  never get run. If you directly specify the path for `cog_type`, the tests will fail because none of the created CollectionObjectGroupTypes have a Collection (which is a required field). 

https://github.com/specify/specify7/blob/76dceab023068707a7ca43d3fa5f86c00510b6a3/specifyweb/businessrules/tests/cog_type.py#L9-L15

Reply:
See https://github.com/specify/specify7/blob/production/specifyweb/specify/tests/__init__.py

Reply:
don't need __init__ mess. rename should do it (to test_blah)

Reply:
Either that or dynamically import all `ApiTests` into the `__init__` namespace. 

Wouldn't have to worry about file names at that point. 

Something like: 
```py
python_files = [
    f for f in os.listdir(os.path.dirname(__file__))
    if f.endswith('.py') and f != '__init__.py'
]

for python_file in python_files:
    module_name = os.path.splitext(python_file)[0]
    
    module = importlib.import_module(f'.{module_name}', __name__)

    for name, obj in inspect.getmembers(module, inspect.isclass):
        if issubclass(obj, ApiTests):
            globals()[name] = obj
```

I'm sure that wouldn't create any problems or anything 😬. 

But yeah, i'll just go with the file renaming route. It's a 'clean' solution: just let UnitTests handle it. 

Since we're on the topic,  I think we can definitely go with a better approach for https://github.com/specify/specify7/blob/production/specifyweb/businessrules/rules/__init__.py. 

Wouldn't `from . import *` just be good enough? I'll take a look

Reply:
maybe if we put that in the init file, it would have. oh wait